bin/minikube/launch.py

```diff
@@ -0,0 +1,1049 @@
+import base64
+import json
+import os
+import secrets
+import subprocess
+import time
+from enum import Enum
+from itertools import chain
+from pathlib import Path
+from typing import Callable, Optional, cast
+
+import typer
+from pydantic import BaseModel
+from toolz.dicttoolz import dissoc
+
+app = typer.Typer()
+
+
+GIT_ROOT = Path(
+    subprocess.check_output(["git", "rev-parse", "--show-toplevel"], text=True).strip()
+)
+
+HELM_CHART_PATH = GIT_ROOT / "helm-chart"
+SAVE_DIRECTORY = GIT_ROOT / "bin" / "minikube" / "build"
+
+
+class AuthKind(Enum):
+    EMAIL = "email"
+    USERNAME = "username"
+
+
+DEFAULT_AVATAR_VERSION = "latest"
+DEFAULT_API_BASE_URL = "http://localhost:8000"
+
+DEFAULT_IS_SENTRY_ENABLED = False
+DEFAULT_IS_TELEMETRY_ENABLED = False
+
+DEFAULT_PDFGENERATOR_VERSION = "latest"
+DEFAULT_AUTHENTICATION_KIND = AuthKind.USERNAME
+DEFAULT_ORGANIZATION_NAME = "octopize"
+DEFAULT_SHARED_STORAGE_PATH = "/home/avatar/shared"
+DEFAULT_SHOULD_USE_LOCAL_STORAGE = True
+
+DEFAULT_DB_NAME = "avatar"
+
+DEFAULT_USERNAME = "avatar_admin"
+DEFAULT_WORKER_MEMORY_REQUEST = "4Gi"
+DEFAULT_API_MEMORY_REQUEST = "1Gi"
+DEFAULT_PDFGENERATOR_MEMORY_REQUEST = "2Gi"
+DEFAULT_WORKER_CPU_REQUEST = "1000m"
+DEFAULT_API_CPU_REQUEST = "512m"
+DEFAULT_PDFGENERATOR_CPU_REQUEST = "512m"
+
+
+class InvalidConfig(Exception):
+    pass
+
+
+class Authentication(BaseModel):
+    kind: AuthKind
```

Comment:
I find it strange semantically to have `Authentication` being only a kind. Here you're not forcing anything, as if a 3rd auth inherited `Authentication`, it would not have to implement `kind`. Maybe use a protocol/interface instead?

Reply:
I think Authentication didn't even need to be a BaseModel, I overengineered that part, I don't think we need to worry any time soon about releasing a new Authentication mechanism, so I'm going to just leave it as is.